x-pack/solutions/security/packages/expandable-flyout/src/hooks/use_window_width.ts

```diff
@@ -33,46 +40,79 @@ export const useWindowWidth = (): number => {
 
       const windowWidth = window.innerWidth;
       if (windowWidth !== 0) {
-        let rightSectionWidth: number;
+        let rightSectionWidthOverlayMode: number;
+        let rightSectionWidthPushMode: number;
         if (windowWidth < MIN_RESOLUTION_BREAKPOINT) {
           // the right section's width will grow from 380px (at 992px resolution) while handling tiny screens by not going smaller than the window width
-          rightSectionWidth = Math.min(RIGHT_SECTION_MIN_WIDTH, windowWidth);
+          rightSectionWidthOverlayMode = Math.min(
+            RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE,
+            windowWidth
+          );
+          rightSectionWidthPushMode = Math.min(RIGHT_SECTION_MIN_WIDTH_PUSH_MODE, windowWidth);
         } else {
-          const ratioWidth =
-            (RIGHT_SECTION_MAX_WIDTH - RIGHT_SECTION_MIN_WIDTH) *
+          const ratioWidthOverlayMode =
+            (RIGHT_SECTION_MAX_WIDTH_OVERLAY_MODE - RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE) *
             ((windowWidth - MIN_RESOLUTION_BREAKPOINT) /
               (MAX_RESOLUTION_BREAKPOINT - MIN_RESOLUTION_BREAKPOINT));
 
           // the right section's width will grow to 750px (at 1920px resolution) and will never go bigger than 750px in higher resolutions
-          rightSectionWidth = Math.min(
-            RIGHT_SECTION_MIN_WIDTH + ratioWidth,
-            RIGHT_SECTION_MAX_WIDTH
+          rightSectionWidthOverlayMode = Math.min(
+            RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE + ratioWidthOverlayMode,
+            RIGHT_SECTION_MAX_WIDTH_OVERLAY_MODE
+          );
+
+          const ratioWidthPushMode =
+            (RIGHT_SECTION_MAX_WIDTH_PUSH_MODE - RIGHT_SECTION_MIN_WIDTH_PUSH_MODE) *
+            ((windowWidth - FULL_WIDTH_BREAKPOINT) /
+              (BIG_RESOLUTION_BREAKPOINT - FULL_WIDTH_BREAKPOINT));
```

Comment:
could you add some comments about `BIG_RESOLUTION_BREAKPOINT`, `FULL_WIDTH_BREAKPOINT`, and their difference from `MIN_RESOLUTION_BREAKPOINT` and `MAX_RESOLUTION_BREAKPOINT`? 

looks like the latter 2 are used exclusively in overlay mode, maybe adding an overlay reference? or put it in the overlay const object if you take the grouping suggestion

Reply:
same comment as I made [above](https://github.com/elastic/kibana/pull/206155#discussion_r1927758114), let me know what you think of the new code!